v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/transforms/AvroWriteTransform.java

```diff
@@ -196,19 +250,26 @@ public WriteFilesResult<AvroDestination> writeToGCS(PCollection<GenericRecord> g
   }
 
   public static class ConvertBytesToGenericRecord
-      extends DoFn<KafkaRecord<byte[], byte[]>, GenericRecord> {
+      extends DoFn<
+          KafkaRecord<byte[], byte[]>,
+          FailsafeElement<KafkaRecord<byte[], byte[]>, GenericRecord>> {
     private transient SchemaRegistryClient schemaRegistryClient;
     private String schemaRegistryURL = null;
     private String schemaPath = null;
     private boolean useMock;
 
-    public ConvertBytesToGenericRecord(String schemaRegistryURL) {
+    private final BadRecordRouter badRecordRouter;
+
+    public ConvertBytesToGenericRecord(String schemaRegistryURL, BadRecordRouter badRecordRouter) {
       this.schemaRegistryURL = schemaRegistryURL;
+      this.badRecordRouter = badRecordRouter;
     }
 
-    public ConvertBytesToGenericRecord(String schemaPath, boolean useMock) {
+    public ConvertBytesToGenericRecord(
+        String schemaPath, boolean useMock, BadRecordRouter badRecordRouter) {
       this.schemaPath = schemaPath;
       this.useMock = useMock;
```

Comment:
I'm not sure I understand how `useMock` is used and why it's needed.

What if someone calls ` new ConvertBytesToGenericRecord(schemaPath, useMock = false, badRecordRouter)`? AFAIU, `setup()` method will try to use `schemaRegistryURL` in this case, which is not initialized.

I'd implement it like this:
```
  public static ConvertBytesToGenericRecord withSchemaRegistry(String schemaRegistryUrl, BadRecordRouter badRecordRouter) {
    return new ConvertBytesToGenericRecord(schemaRegistryUrl, null, badRecordRouter);
  }

  public static ConvertBytesToGenericRecord withSchemaPath(String schemaPath, BadRecordRouter badRecordRouter) {
    return new ConvertBytesToGenericRecord(null, schemaPath, badRecordRouter);
  }

  private ConvertBytesToGenericRecord(String schemaRegistryUrl, String schemaPath, BadRecordRouter badRecordRouter) {
      this.schemaRegistryUrl = schemaRegistryUrl;
      this.schemaPath = schemaPath;
      this.badRecordRouter = badRecordRouter;
    }
```

Reply:
Added a TODO to remove mock